x-pack/plugins/security_solution/public/timelines/components/side_panel/event_details/index.tsx

```diff
@@ -224,18 +250,62 @@ const EventDetailsPanelComponent: React.FC<EventDetailsPanelProps> = ({
             hostRisk={hostRisk}
             handleOnEventClosed={handleOnEventClosed}
           />
-        )}
-      </StyledEuiFlyoutBody>
+        );
+    }
+  }, [
+    agentId,
+    browserFields,
+    detailsData,
+    expandedEvent,
+    handleIsolationActionSuccess,
+    handleOnEventClosed,
+    hostRisk,
+    isActivePanel,
+    isAlert,
+    isDraggable,
+    isolateAction,
+    loading,
+    osquery,
+    rawEventData,
+    showAlertDetails,
+    timelineId,
+  ]);
+
+  if (!expandedEvent?.eventId) {
+    return null;
+  }
+
+  const handlePanelChange = (panelType: ACTIVE_PANEL | null) => {
+    if (isActivePanel === ACTIVE_PANEL.OSQUERY && panelType === null) {
+      showAlertDetails();
+      return;
```

Comment:
more of a style preference, but mind using an `else` here?

Reply:
I don't mind at all, changed 👍